src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,164 @@
+                            + "bigger than the actual size of list");
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(value, first, null);
+            first.prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> currentFirstNode = findNode(index - 1);
+            Node<T> currentLastNode = currentFirstNode.next;
+            currentFirstNode.next = new Node<>(value, currentLastNode, currentFirstNode);
+            currentLastNode.prev = currentFirstNode.next;
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexSuits(index);
+        Node<T> currentNode = findNode(index);
+        return currentNode.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isIndexSuits(index);
+        Node<T> currentNode = findNode(index);
+        T returnElement = currentNode.element;
+        currentNode.element = value;
+        return returnElement;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexSuits(index);
+        if (index == 0) {
+            return removeFirstElement();
+        }
+        if (index == size - 1) {
+            return removeLastElement();
+        }
+        Node<T> currentNode = findNode(index);
+        Node<T> cupNode = currentNode.prev;
+        cupNode.next = currentNode.next;
+        cupNode = currentNode.next;
+        cupNode.prev = currentNode.prev;
+        size--;
+        return currentNode.element;
     }
 
     @Override
     public boolean remove(T t) {
+        for (int i = 0; i < size; i++) {
+            if (get(i) == t || get(i) != null && get(i).equals(t)) {
+                remove(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T removeLastElement() {
+        final T removedValue = last.element;
+        last = last.prev;
+        last.next = null;
+        size--;
+        return removedValue;
+    }
+
+    private T removeFirstElement() {
+        T removedValue = first.element;
+        if (size == 1) {
+            first = null;
+            last = null;
+            size--;
+            return removedValue;
+        }
+        first = first.next;
+        first.prev = null;
+        size--;
+        return removedValue;
+    }
+
+    private Node<T> findNode(int index) {
+        int currentSize = 0;
+        Node<T> currentNode = first;
```

Comment:
We need to use shorter path, so let's go from last element to middle of from first element to middle depending on our index